htdocs/core/lib/company.lib.php

```diff
@@ -1084,8 +1084,46 @@ function show_actions_done($conf, $langs, $db, $filterobj, $objcon='', $noprint=
             if ($filterobj->id) $sql.= " AND a.fk_element = ".$filterobj->id;
         }
         if (is_object($objcon) && $objcon->id) $sql.= " AND a.fk_contact = ".$objcon->id;
+        
+        $agendaDefaultFilter = unserialize($conf->global->AGENDA_DEFAULT_FILTER_VIEW);
         // Condition on actioncode
-        if (! empty($actioncode))
+        if ($actioncode == '' && !empty($agendaDefaultFilter)){
+            
+            if (empty($conf->global->AGENDA_USE_EVENT_TYPE))
+            {
+                if ((in_array('AC_NON_AUTO', $agendaDefaultFilter) || in_array('AC_OTH', $agendaDefaultFilter)) && (in_array('AC_OTH_AUTO', $agendaDefaultFilter) || in_array('AC_ALL_AUTO', $agendaDefaultFilter))){
+                    // do nothing
+                }  elseif (in_array('AC_NON_AUTO', $agendaDefaultFilter) || in_array('AC_OTH', $agendaDefaultFilter)) {
+                    $sql.= " AND c.type != 'systemauto'";
+                } elseif (in_array('AC_OTH_AUTO', $agendaDefaultFilter) || in_array('AC_ALL_AUTO', $agendaDefaultFilter)) {
+                    $sql.= " AND c.type = 'systemauto'";
+                }
+            }
+            else
+            {
+                if (in_array('AC_NON_AUTO', $agendaDefaultFilter) && in_array('AC_ALL_AUTO', $agendaDefaultFilter)){
+                    // do nothing
+                }  elseif (in_array('AC_NON_AUTO', $agendaDefaultFilter)) {
+                    $sql.= " AND c.type != 'systemauto'";
+                } elseif (in_array('AC_ALL_AUTO', $agendaDefaultFilter)) {
+                    $sql.= " AND c.type = 'systemauto'";
+                }
+                
+                $arraydone = array('AC_NON_AUTO', 'AC_ALL_AUTO');
+                $i = 0;
+                foreach ($agendaDefaultFilter as $k => $ac) {
+                    if (!in_array($ac, $arraydone)) {
+                        if (empty($i)){
+                            $sql.= " AND (c.code = '".$db->escape($ac)."'";$i++;
+                        }
+                        else $sql.= " OR c.code = '".$db->escape($ac)."'";
+                        
+                    }
+                }
+                
+                if(!empty($i)) $sql.=")";
```

Comment:
Donc là tu mets la parenthèse si $i défini alors que avant tu as fais le test si vide et dans la boucle while. Je comprends pas

Reply:
en fait s'il est empty c'est qu'on n'a pas rajouté de 1er critère de code... donc pas de "AND (c.code = "
Du coup, pas besoin de fermer la parenthèse qui n'a pas été ouverte.

Reply:
Exemple typique où un dev ce suicide en repassant dessus

pour simplifier, @atm-greg , voila ce qui serait plus saint :
`$TOrSql = array();`
`foreach ($agendaDefaultFilter as $k => $ac) {  if (!in_array($ac, $arraydone)) {  $TOrSql[] = " c.code = '".$db->escape($ac)."'";  }  }`

`if (!empty($TOrSql)) {  $sql .=  " AND (".implode(' OR ',  $TOrSql).")"; }`